### PR TITLE
[DPM-791] support bonus deposit

### PR DESCRIPTION
**Pull request by alrxy**

### CODE REVIEW

File: blockchain.go [hunk 1]
```diff
@@ -137,14 +149,77 @@ func ValidateSignatures(pubKeys []ecc.PublicKey, platformPubKey ecc.PublicKey) e
...
+	return invariant, nil
+}
+
+func isInvariantAllowed(invariant []string) bool {
+	for _, inv := range allowedInvariants {
+		if len(inv) != len(invariant) {
+			continue
+		}
+
+		isBreak := false
```
Comment (justefg) [inline blockchain.go]: i think `matches := true` would be a better name

Comment (alrxy) [reply blockchain.go]: ok

**Review by algys — APPROVED** [review]

File: blockchain.go [hunk 2]
```diff
@@ -11,6 +11,18 @@ import (
...
 
+var (
+	allowedInvariants = [][]string{
+		{"transfer", "newgame"},
+		{"transfer", "newgame", "gameaction"},
+		{"transfer", "gameaction"},
+		{"transfer", "newgamebon", "gameaction"},
+		{"newgamebon", "gameaction"},
+		{"transfer", "depositbon", "gameaction"},
+		{"depositbon", "gameaction"},
```
Comment (justefg) [inline blockchain.go:22]: shouldn't be allowed

Comment (alrxy) [reply blockchain.go:22]: all 4?

**Review by justefg — APPROVED** [review]

**Review by justefg — APPROVED** [review]

**Pull request merged**